docs/guides/how_to_guides/creating_batches/how_to_load_a_pandas_dataframe_as_a_batch.rst

```diff
@@ -1,18 +1,145 @@
 .. _how_to_guides__creating_batches__how_to_load_a_pandas_dataframe_as_a_batch:
 
-How to load a Pandas dataframe as a Batch
+How to load a Pandas DataFrame as a Batch
 =========================================
 
-.. admonition:: Admonition from Mr. Dickens
+This guide will help you load a Pandas DataFrame as a Batch for use in creating expectations.
 
-    "Whether I shall turn out to be the hero of my own life, or whether that station will be held by anybody else, these pages must show."
+.. content-tabs::
 
+    .. tab-container:: tab0
+        :title: Show Docs for Stable API (up to 0.12.x)
 
-This guide is a stub. We all know that it will be useful, but no one has made time to write it yet.
+        .. admonition:: Prerequisites: This how-to guide assumes you have already:
 
-If it would be useful to you, please comment with a +1 and feel free to add any suggestions or questions below.
+            - :ref:`Set up a working deployment of Great Expectations <tutorials__getting_started>`
+            - Configured a :ref:`Pandas/filesystem Datasource <how_to_guides__configuring_datasources>`
+            - Identified a pandas DataFrame that you would like to use as the data to validate.
+
+        Steps
+        -----
+
+        1. Construct batch_kwargs
+
+            batch_kwargs describe the data you plan to validate. Here we are using a datasource you have configured and passing in a DataFrame under the ``"dataset"`` key.
+
+            .. code-block:: python
+
+                batch_kwargs = {
+                    "datasource": "insert_your_datasource_name_here",
+                    "dataset": insert_your_dataframe_here
```

Comment:
Very cool - I did not know about the `ge.read_csv()` path. I'm thinking how about describing both, and I agree to defaulting to the `read_csv()` path. The reason for describing both is if someone has a dataframe created from some other method than `read_csv()` and want to pass it in directly.

Reply:
Oh true, good point, you'll want to describe the non-read-csv dataframe usecase too. Makes sense to describe both, with option 1 being the simple method, and option 2 being more complex.

Reply:
Let me know how you like the two paths in the old API, happy to modify if there is a better way to present it 😄

Reply:
@spbail @anthonyburdi ge.read_csv does not work with DataContext, because it is an ancient pre-DataContext method. The object that it returns cannot be used the same way as a batch obtained from a DataContext (e.g., if you call `batch.save_expectation_suite(discard_failed_expectations=False)` on it, it will throw `ValueError: Unable to save config: filepath or data_context must be available.`.

Let's not mention this method in this guide.

Reply:
Aah ok, thanks for the heads up @eugmandel.

Reply:
Sounds good! I'll remove it tomorrow.